docs/azure/services/Azure_Functions.md

```diff
@@ -0,0 +1,109 @@
+
+The assessment notes on best security practices for NSGs can be found [here](https://learn.microsoft.com/en-us/azure/virtual-network/network-security-groups-overview) and for UDRs [here]( https://learn.microsoft.com/en-us/azure/virtual-network/vnet-integration-for-azure-services). However, it is worth noting that inbound rules will not affect the inbound access to the function, which is controlled by the access restrictions at the service-level configuration that were previous described.
+
+Finally, when VNet integration is set up, no outbound restrictions are set by default. This means that that the function can communicate to the internet. To limit this, a “Route All” feature can be enabled, forcing a traffic originating from the function to comply with the UDRs and then specified NSGs which should block all unnecessary internet access.
+
+### Encryption in Transit
+Azure Functions encrypts all the incoming data using TLS 1.2 by default. However, the service does allow for manual downgrading. When reviewing the configuration, users must be ensure that TLS 1.2 is enforced for all communication to the deployed function.
+
+Furthermore, clients connecting to the function can opt to use either HTTP or HTTPS. Where possible, any HTTP traffic should be redirected to HTTPS. Reviewing whether the latest version of TLS and HTTPS only traffic has been enforced, can be done either in the `TLS/SSL` settings blade or via the following CLI command:
+```CLI
+az functionapp show --name MyFunctionApp --resource-group MyResourceGroup --query “httpsOnly”
+az functionapp show --name MyFunctionApp --resource-group MyResourceGroup --query "minTlsVersion"
+```
+Azure Functions have an FTP endpoint that is enabled by default. If not used, this endpoint should be disabled to restrict the possible ways the function app can be interacted with. If required as part of business operation, deployments should be forced to use FTPS. To check the FTP configuration, the following command could be run:
+```CLI
+az functionapp show --name AzFunTestBl --resource-group AzFun --query siteConfig."ftpsState"
+```
+
+### Function Access Keys
+Unless a HTTP-triggered function is set to “anonymous”, two authorisation levels configured on creation determine whether the function app is using a “function-level” or “admin-level” access key. 
+
+Function-level access key can authorise against a specific function, whilst admin-level uses a “master-key” for authentication against all functions in the Function App. Each Function App has a master-key on creation which can’t be removed. Additionally, it also provides administrative access to the runtime’s APIs. Because of the authorisation scope that the master key provides, users should be careful with how it is handled and distributed as part of their secrets management process.
```

Comment:
Well we're missing out host keys in regards to authorisation scopes (function-keys). Host keys allow a user to access all functions in the function app. The master key (admin-level) not only does it give access to all functions in the app but it also provides administrative access to the REST APIs and more importantly the key can't be revoked. So its important we make it a clear distinction.